pkg/datapath/linux/probes/probes.go

```diff
@@ -312,31 +305,11 @@ func (p *ProbeManager) GetOptionalConfig() map[KernelParam]kernelOption {
 }
 
 // GetMapTypes returns information about supported BPF map types.
+// Deprecated. Use HaveMapType() instead.
```

Comment:
The only remaining user of this method seems to be in `probes_privileged_test.go` for `TestMapTypes`. The purpose of that test seems to be testing `GetMapTypes`. I think we can thus remove the test and this method as well. While at it maybe also add a test for `HaveMapType` instead? Can both be done as a follow-up PR.

Reply:
> The only remaining user of this method seems to be in probes_privileged_test.go for TestMapTypes. The purpose of that test seems to be testing GetMapTypes. I think we can thus remove the test and this method as well. 

Good point, done!

> While at it maybe also add a test for HaveMapType instead?

Package `features` already has an extensive test suite in the lib itself, so that should be covered. The specific map/prog types the Cilium agent probes for were even used to choose the initial batch of handcrafted tests that are expected to pass on specific kernel versions we need to support!

`HaveMapType` and friends are simple logging wrappers that can't really be tested.